lib/addserver.dart

```diff
@@ -0,0 +1,268 @@
+import 'dart:convert';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'dart:io';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'hosts.dart';
+import 'server_list.dart';
+class AddHost extends StatefulWidget {
+  const AddHost({Key? key, required this.title, this.host}) : super(key: key);
+
+  final String title;
+  final Hosts? host; // Add the host parameter
+
+  @override
+  State<AddHost> createState() => _AddHostState();
+}
+
+class _AddHostState extends State<AddHost> {
+  final int port = 9;
+  String macAddress = '';
+  String ipAddress = '';
+  String hostName = '';
+  Hosts? host; // Add the host variable
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.host != null) {
+      // Initialize fields with the provided host object
+      host = widget.host;
+      hostName = host!.hostName;
+      ipAddress = host!.ipAddress;
+      macAddress = host!.macAddress;
+    }
+    loadPreferences();
+  }
+
+//using json to manage listed objects.
+  Future<void> savePreferences() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```

Comment:
Declare late variable on top of state and initialize in initState, then reuse variable.